File: 2023-2024/sourcecode/mod3/Lab7/MVVMBasicsJava/app/src/main/java/com/example/mvvmbasicsjava/ui/fragment/UserComparator.java
package com.example.mvvmbasicsjava.ui.fragment;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import com.example.mvvmbasicsjava.model.User;

public class UserComparator extends DiffUtil.ItemCallback<User> {
    @Override
    public boolean areItemsTheSame(@NonNull User oldItem, @NonNull User newItem) {
        return oldItem == newItem;
    }

    @Override
    public boolean areContentsTheSame(User oldItem, @NonNull User newItem) {
        return oldItem.equals(newItem);
    }
}

File: 2023-2024/sourcecode/mod3/Lab7/MVVMBasicsKotlin/app/src/main/java/com/example/mvvmbasicskotlin/ui/fragment/UserComparator.kt
package com.example.mvvmbasicskotlin.ui.fragment

import androidx.recyclerview.widget.DiffUtil
import com.example.mvvmbasicskotlin.model.User

class UserComparator : DiffUtil.ItemCallback<User>() {
    override fun areItemsTheSame(oldItem: User, newItem: User): Boolean {
        return oldItem === newItem
    }

    override fun areContentsTheSame(oldItem: User, newItem: User): Boolean {
        return oldItem == newItem
    }
}